lib/graphql/static_validation/rules/variable_usages_are_allowed.rb

```diff
@@ -7,32 +7,35 @@ class VariableUsagesAreAllowed
       def validate(context)
         # holds { name => ast_node } pairs
         declared_variables = {}
-
         context.visitor[GraphQL::Language::Nodes::OperationDefinition] << ->(node, parent) {
           declared_variables = node.variables.each_with_object({}) { |var, memo| memo[var.name] = var }
         }
 
         context.visitor[GraphQL::Language::Nodes::Argument] << ->(node, parent) {
-          return if !node.value.is_a?(GraphQL::Language::Nodes::VariableIdentifier)
-          arguments = nil
-
-          case parent
-          when GraphQL::Language::Nodes::Field
-            arguments = context.field_definition.arguments
-          when GraphQL::Language::Nodes::Directive
-            arguments = context.directive_definition.arguments
-          when GraphQL::Language::Nodes::InputObject
-            arg_type = context.argument_definition.type.unwrap
-            if arg_type.is_a?(GraphQL::InputObjectType)
-              arguments = arg_type.input_fields
+          node_values = Array.wrap(node.value)
+          return if !(node_values.all? { |value| value.is_a? GraphQL::Language::Nodes::VariableIdentifier })
```

Comment:
What happens if the query string is like this: 

```graphql 
query GetStuff($int1: Int, $int2: Int) {
  stuff(ints: [1, 2, $int1, $int2])
}
```

where there is a _mix_ of literal values (`1, 2`) and variables (`$int1, $int2`)? I think it would fail this check, since they aren't _all_ variables. 

I wonder, could we still validate those by _filtering_ the node values: 

```ruby 
variable_values = node_values.select { |n| n.is_a?(GraphQL::Language::Nodes::VariableIdentifier) }
if variable_values.any? 
  # ... 
  variable_values.each do ... 
```

Do you think that would work here?